### DISCUSSION TITLE
Conda recipe

### DISCUSSION THREAD

**Discussion by sarthakpati** [TorchIO-project/torchio#480] [Ideas]
Hey Folks,

I have added a conda recipe for TorchIO and I have put in a PR at [conda-forge](https://github.com/conda-forge/staged-recipes/pull/14200). It'd be great if @fepegar or other maintainers could post a comment there so that it could get accepted and merged!

Cheers,
Sarthak

**Comment by fepegar** [comment]
That's great! Thanks, @sarthakpati. What would you like me to comment?

**Reply by sarthakpati** [reply]
Yes, please! I am still having some weird issues with their CI, but let's see.

**Reply by fepegar** [reply]
I'm still not sure what you would like me to say.

Also, can you provide some info about the recipe? Why is it necessary? Will it need maintenance?

**Reply by sarthakpati** [reply]
> I'm still not sure what you would like me to say.

Just a single comment that you are the maintainer of this repo, i.e., https://github.com/fepegar/torchio

> Also, can you provide some info about the recipe?

https://github.com/sarthakpati/staged-recipes/blob/torchio/recipes/torchio/meta.yaml

> Why is it necessary? 

Because one of the packages that I am working on depends on TorchIO, and since the only way for me to disseminate mine is via conda (it has C components), I need to make TorchIO available on conda before I can push my package's recipe out.

> Will it need maintenance?

Nope, because it will be pulled from PyPi using `conda skeleton`. If it needs anything extra, I am happy to provide that support because I have a vested interest in having TorchIO available on conda. 😄

**Reply by fepegar** [reply]
Ok, thanks for the explanation. I see in that YAML a specific PyPI version. Won't we need to update that every time a new version of TorchIO is released?

**Reply by sarthakpati** [reply]
I would imagine, but I am happy to take that on. It is also the first time I am sending a PyPi recipe to conda, so it is a learning experience for me.

**Reply by fepegar** [reply]
I guess you want to comply with

> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

I suggest that you just list yourself as the maintainer instead.

**Reply by sarthakpati** [reply]
Alright.

**Reply by sarthakpati** [reply]
It is ready to merged in. Once that is done, can I put a PR to the [installation instructions](https://torchio.readthedocs.io/quickstart.html#installation) to allow people to install from conda, as well?

**Reply by fepegar** [reply]
Thanks @sarthakpati, but I think we'd better wait until we are sure that the recipes are being updated automatically after each release, as it's done for `pip`.

**Reply by sarthakpati** [reply]
I guess https://github.com/regro-cf-autotick-bot is what does this "auto version update". I am not 100% sure, as this is the first time I am putting something on conda-forge. Looking at [SimpleITK's conda-forge PR feed](orge/simpleitk-feedstock/pulls?q=is%3Apr+is%3Aclosed), it seems to be working as expected.

**Reply by sarthakpati** [reply]
FYI: 
![image](https://user-images.githubusercontent.com/11719673/110795933-61cb4d80-8245-11eb-8ba5-c8200b5418e5.png)

**Reply by fepegar** [reply]
Thanks. Looks promising. If we see that the next releases are properly updated, we can start thinking about documenting this installation method.

**Reply by sarthakpati** [reply]
Great, thanks.

**Reply by sarthakpati** [reply]
Available now: https://github.com/conda-forge/torchio-feedstock

https://anaconda.org/conda-forge/torchio

**Reply by fepegar** [reply]
Nice. Thanks @sarthakpati!

**Reply by sarthakpati** [reply]
As [this PR](https://github.com/conda-forge/torchio-feedstock/pull/1) shows, the regro-cf-autotick-bot automatically updates the version on conda (subject to getting merged, of course). I would like to request https://github.com/fepegar/torchio/pull/485 be reopened, and perhaps additional documentation could be added (including my contribution to this project). Cheers!

**Reply by fepegar** [reply]
SimpleITK 2 + multiprocessing seems to be problematic (https://github.com/SimpleITK/SimpleITK/issues/1239). Is it possible to add the `SimpleITK<2` requirement to the recipe?

**Reply by sarthakpati** [reply]
Updated: https://github.com/conda-forge/torchio-feedstock/blob/master/recipe/meta.yaml#L33
